inc/commondbtm.class.php

```diff
@@ -1391,19 +1391,77 @@ function prepareInputForAdd($input) {
       return $input;
    }
 
-    /**
+   /**
+    * Compute the name of a copied item
+    *
+    * @param string   $current_item The item being copied
+    * @param null|int $copy_index   The index to append to the name
+    *
+    * @return string The computed name of the new item to be created
+    */
+   public function computeCloneName(
+      string $current_name,
+      ?int $copy_index = null
+   ): string {
+      // No index specified or first copy
+      if (is_null($copy_index) || $copy_index == 1) {
+         return sprintf(__("%s (copy)"), $current_name);
+      }
+
+      // 1+ copy, add index
+      return sprintf(__("%s (copy %d)"), $current_name, $copy_index);
+   }
+
+   /**
     * Prepare input datas for cloning the item
     *
     * @since 9.5
     *
     * @param array $input datas used to add the item
     *
     * @return array the modified $input array
-   **/
+    */
    function prepareInputForClone($input) {
+      global $DB;
+
       unset($input['id']);
       unset($input['date_mod']);
       unset($input['date_creation']);
+
+      // Force uniqueness for the name field
+      if (isset($input[static::getNameField()])) {
+         $current_name = $input[static::getNameField()];
+         $copy_name = "";
+
+         // Find all exisiting names starting like our current item
+         $data = $DB->request([
+            'SELECT' => [static::getNameField()],
+            'FROM'   => $this->getTable(),
+            'WHERE'  => [
+               static::getNameField() => ['LIKE', "$current_name%"]
+            ]
+         ]);
```

Comment:
In some languages, the base name may be at a different position, so assuming that it is always placed at the beginning of the computed clone name is not correct.
Also, the `computeCloneName` method can be overridden and produce something that may not exactly contains the base name (i.e. cloning `test-----user` will produce `test-user-copy`).

So I guess you have no choice than using a `while (countElementsInTable($table, [$namefield => $copy_name]) > 0)` condition.

Reply:
Yeah I've tried to avoid doing this but I guess it's not possible to avoid it.

I'll look into having some kind of cache, because with this solution if you clone an item `n` times you will have at least (if all copy names are available) `n * (n +1) /2` queries (because each copy will start over it's search from the first 'copy index').
This means if you clone an item 60 times you will run 1830 queries.

I suppose it's not a very common usage but It might be better to think ahead and avoid crashing the database if someone actually tries it.